packages/driver/src/cross-origin/communicator.ts

```diff
@@ -242,15 +274,16 @@ export class SpecBridgeCommunicator extends EventEmitter {
   onMessage ({ data }) {
     if (!data) return
 
-    this.emit(data.event, data.data)
+    this.emit(data.event, data.data, { responseEvent: data.responseEvent })
   }
 
   /**
    * Events to be sent to the primary communicator instance.
    * @param {string} event - the name of the event to be sent.
    * @param {Cypress.ObjectLike} data - any meta data to be sent with the event.
+   * @param responseEvent - the even to be responded with when sending back a result.
    */
-  toPrimary (event: string, data?: Cypress.ObjectLike, options: { syncGlobals: boolean } = { syncGlobals: false }) {
+  toPrimary (event: string, data?: Cypress.ObjectLike, options: { syncGlobals: boolean } = { syncGlobals: false }, responseEvent?: string) {
```

Comment:
we should consider making toPrimary take an object too. I didn't make the change here because i was already making enough changes, but would be willing to update if everyone agrees.